src/dtemplate.d

```diff
@@ -4105,6 +4109,44 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
                 result = match2;
                 return;
             }
+
+            /* Deduce common type
```

Comment:
I changed code to add a new function `rawTypeMerge` in `dcast.d` and use it here.